packages/data-stores/src/user/actions.ts

```diff
@@ -2,49 +2,57 @@
  * Internal dependencies
  */
 import {
-	ActionType,
 	CurrentUser,
 	CreateAccountParams,
 	NewUserErrorResponse,
 	NewUserSuccessResponse,
 } from './types';
 
 export const fetchCurrentUser = () => ( {
-	type: ActionType.FETCH_CURRENT_USER as const,
+	type: 'FETCH_CURRENT_USER' as const,
 } );
 
 export const receiveCurrentUser = ( currentUser: CurrentUser ) => ( {
-	type: ActionType.RECEIVE_CURRENT_USER as const,
+	type: 'RECEIVE_CURRENT_USER' as const,
 	currentUser,
 } );
 
 export const receiveCurrentUserFailed = () => ( {
-	type: ActionType.RECEIVE_CURRENT_USER_FAILED as const,
+	type: 'RECEIVE_CURRENT_USER_FAILED' as const,
 } );
 
 export const fetchNewUser = () => ( {
-	type: ActionType.FETCH_NEW_USER as const,
+	type: 'FETCH_NEW_USER' as const,
 } );
 
 export const receiveNewUser = ( response: NewUserSuccessResponse ) => ( {
-	type: ActionType.RECEIVE_NEW_USER as const,
+	type: 'RECEIVE_NEW_USER' as const,
 	response,
 } );
 
 export const receiveNewUserFailed = ( error: NewUserErrorResponse ) => ( {
-	type: ActionType.RECEIVE_NEW_USER_FAILED as const,
+	type: 'RECEIVE_NEW_USER_FAILED' as const,
 	error,
 } );
 
 export function* createAccount( params: CreateAccountParams ) {
 	yield fetchNewUser();
 	try {
 		const newUser = yield {
-			type: ActionType.CREATE_ACCOUNT as const,
+			type: 'CREATE_ACCOUNT' as const,
 			params,
 		};
 		return receiveNewUser( newUser );
 	} catch ( err ) {
 		return receiveNewUserFailed( err );
 	}
 }
```

Comment:
When working on this, I was curious about this.

- There's an untyped literal action object that's yielded in the middle. An observation, not necessarily an issue.
- This is a generator, which uniquely types its yield and return types.

I'm not sure whether our reducers "see" yielded actions (presumably handled by controls?) or only returned actions. Maybe this is up to the handling control and not something we can generalize about.

Reply:
Just tried this and was surprised to see that the reducer _didn't_ see the `'CREATE_ACCOUNT'` action 🤔

Reply:
@p-jackson I ran into this when I was working on the Site data store, and it looks like the way the [redux-routine middleware](https://developer.wordpress.org/block-editor/packages/packages-redux-routine/) works is that if it matches a control against the name of an action, it [runs the control](https://github.com/WordPress/gutenberg/blob/5920ab8620e0a5ea6ed810e399dc42fd94d99abd/packages/redux-routine/src/runtime.js#L26) but _doesn't_ dispatch the action to pass it along to the reducer. For unhandled actions (actions that don't have a match in the controls), they [do get dispatched](https://github.com/WordPress/gutenberg/blob/5920ab8620e0a5ea6ed810e399dc42fd94d99abd/packages/redux-routine/src/runtime.js#L40) by the middleware so that they hit the reducer as normal.

I still don't think I have a solid grasp of exactly how the middleware works, but in effect, it feels like controls are designed to intercept an action, and then in the control we can then dispatch actions that will hit the reducer. Because the reducer doesn't see the action that's listed in our controls, that's why we've got the additional `fetchNewUser` action that gets yielded before `CREATE_ACCOUNT`, otherwise from the reducer, we don't know that the action has started. It seems like useful behaviour for a middleware to me, but definitely non-obvious in its behaviour!